Save data

zbahati opened the pull request:
- Create methods  to save data into books, people, and rentals)using JSON files. 
- On startup of app  the system load all data from files.

#Finishes

Errytagedesign committed:
add datamanager.rb

Errytagedesign committed:
require data_manager in main.rb

Errytagedesign committed:
Add menu_data dir, add list_all_books.rb

Errytagedesign committed:
Update person

Commit:
Ft save Data

- Add Methods
- Add all_rentals_rb file to retrieve data from rentals.json
-  fix rubocop errors

# Finishes ..

Johnadibe commented on people.json:
```diff
@@ -0,0 +1 @@
+[{"age":21,"name":"Ade","parent_permission":true,"id":509},{"age":13,"name":"Sane","parent_permission":true,"id":191},{"age":20,"name":"Bahati","parent_permission":true,"id":989},{"age":25,"name":"Bahati","parent_permission":true,"id":420},{"age":20,"name":"Bahati","parent_permission":true,"id":609},{"age":24,"name":"Alex","parent_permission":true,"id":308},{"age":45,"name":"RE","parent_permissio ...
```
> - [ ] Good job guys👍, Based on the input data we will save, I recommend adding the `"type": teacher or student` field and include `"specialization": ""` for the Teacher class since this information is part of the input data you receive.😊
> 
> Here's an example to illustrate this:
> 
> In your people.json file, the data should be saved in this format:
> 
> ```
> [{"age":21,"name":"Ade","parent_permission":true,"id":509,"type":"student"},{"age":13,"name":"Sane","parent_permission":true,"id":191,"type":"teacher","specialization":"computer"},{"age":20,"name":"Bahati","parent_permission":true,"id":989,"type":"student"},{"age":25,"name":"Bahati","parent_permission":true,"id":420},{"age":20,"name":"Bahati","parent_permission":true,"id":609},{"age":24,"name":"Alex","parent_permission":true,"id":308},{"age":45,"name":"RE","parent_permission":true,"id":557}]
> ```

Johnadibe reviewed (changes requested):
# STATUS: REQUIRED CHANGES ♻️
### Hi @zbahati and @Errytagedesign👋🏽, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

![mario_ladders](https://user-images.githubusercontent.com/20563252/121369265-4e418880-c901-11eb-848c-1eff06d5a4cf.gif)

## Highlights
✅ You used GitFlow correctly 👍🏽
✅ Linter checks passed✔️
✅ One per type of data is stored and named accordingly✔
✅ All data are loaded on startup of the app from files✔️
✅ Handled the case when one or more of the files are missing✔

## Required Changes ♻️

> - [ ] Kindly add the necessary details of the second author who also contributed to the project. This will make your Readme file to be more professional and detailed.👍
> 
> ![auth](https://github.com/zbahati/school_library_oop/assets/40009354/03c54806-ea46-4af9-b8df-ea40f139ce9a)

_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

Commit:
Ft  saveData

Add type and specialization,
Initialization fo type and specialization
update the Readme file

Commit:
Fix Rubocop error

Whoistolu reviewed (approved):
### Hi Team,

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 

# STATUS: APPROVED 🟢
**Congratulations! 🎉** 

### Optional suggestions
Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better.

- [ ] [OPTIONAL] Kindly remove the Live demo section of the README  since this project is yet to be deployed and it is not a requirement to deploy yet.

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @whoistolu in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
